### PR TITLE
Moved documentation to header files

### PR DESCRIPTION
**Pull request by apfohl**
- Moved all function documentation to the corresponding header file
- Reordered functions in header files to match ordering in implementation files
- Added missing documentation
- Leave documentation stubs for missing documentation

### CODE REVIEW

**Comment by apfohl** [comment]
This PR fixed #57.

File: include/libcgi/cgi.h [hunk 1]
```diff
@@ -95,48 +671,147 @@ extern formvars *sess_list_start;
...
+ *
+ * @see https://github.com/rafaelsteil/libcgi/issues/25
+ */
+void cgi_session_free( void );
+
+/**
+ * ToDo: Check where this is coming from as it has no implementation
+ * @param seconds
+ */
+extern void cgi_session_set_max_idle_time(unsigned long seconds);
```
Comment (apfohl) [inline include/libcgi/cgi.h]: I discovered that this function prototype does not have a matching implementation. Is this still needed?

**Comment by apfohl** [comment]
Are there any news on this one? Is this okay? Shall I change something?

**Comment by LeSpocky** [comment]
It's on my ToDo-List, will review it.

File: include/libcgi/cgi_types.h [hunk 2]
```diff
@@ -15,9 +15,9 @@
 #define CGI_TYPES_H
 
 /**
- *	HTTP status codes.
+ * HTTP status codes.
  *
- *	@see	http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html
+ * @see http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html
```
Comment (LeSpocky) [inline include/libcgi/cgi_types.h]: Whitespace change only, right?

File: include/libcgi/cgi_types.h [hunk 3]
```diff
@@ -71,9 +71,9 @@ extern "C" {
 #endif
 
 /**
- *	General purpose linked list. Actually isn't very portable because
- *	uses only 'name' and 'value' variables to store data. Probably, in
- *	a future release, this will be replaced by another type of struct.
+ * General purpose linked list. Actually isn't very portable because
+ * uses only 'name' and 'value' variables to store data. Probably, in
+ * a future release, this will be replaced by another type of struct.
```
Comment (LeSpocky) [inline include/libcgi/cgi_types.h]: This also looks like whitespace change only. Whitespace changes should go to separate commits.

File: include/libcgi/error.h [hunk 4]
```diff
@@ -12,16 +12,34 @@
...
+ */
 
 #define E_WARNING	0
 #define E_FATAL		1
 #define E_CAUTION 	2
 #define E_INFORMATION 	3
 #define E_MEMORY	4
 
+/**
+ * ToDo: Add documentation
```
Comment (LeSpocky) [inline include/libcgi/error.h]: Use `@todo` or `TODO` please.

**Comment by apfohl** [comment]
Yes, of cause I can restructure this PR. Give me some time.
